core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -228,6 +226,52 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+
+		pendingNonce, err := ec.getPendingNonce(ctx, from)
+		if err != nil {
+			return errors.Wrapf(err, "unable to fetch pending nonce for address: %v", from)
+		}
+
+		likelyConfirmed := make([]models.EthTxAttempt, 0)
+		for i := 0; i < len(attempts); i++ {
+			if attempts[i].EthTx.Nonce != nil && *attempts[i].EthTx.Nonce >= int64(pendingNonce) {
```

Comment:
No attempts will ever have a nonce greater than the pending nonce... so this seems pointless.

Reply:
changed to latest nonce